src/components/gauge/gauge.tsx

```diff
@@ -0,0 +1,381 @@
+              !this.styleShadow
+                ? "none"
+                : "",
+            height: `${this.minorSize * 0.8 +
+              (this.calcThickness() * (this.minorSize / 100) -
+                this.minorSize / 100)}px`,
+            width: `${this.minorSize * 0.8 +
+              (this.calcThickness() * (this.minorSize / 100) -
+                this.minorSize / 100)}px`
+          }}
+        />
+        {this.showValue ? (
+          <span
+            class="marker"
+            style={{
+              display: this.showValue ? "" : "none",
+              height: `${this.minorSize * 0.795 -
+                this.calcThickness() * (this.minorSize / 100)}px`,
+              transform:
+                this.calcPercentage() >= 100
+                  ? "rotate(359deg)"
+                  : this.calcPercentage() > 0
+                    ? `rotate(${3.6 * this.calcPercentage()}deg)`
+                    : "rotate(0.5deg)"
+            }}
+          >
+            <div
+              class="indicator"
+              style={{
+                "box-shadow":
+                  (this.minorSize <= 300 && this.thickness <= 25) ||
+                  !this.styleShadow
+                    ? "none"
+                    : "",
+                height:
+                  this.minorSize / 100 +
+                  this.calcThickness() * (this.minorSize / 150) +
+                  "px",
+                transform:
+                  "translateY(-" +
+                  (this.minorSize / 100 +
+                    this.calcThickness() * (this.minorSize / 150)) +
+                  "px)"
+              }}
+            />
+          </span>
+        ) : (
+          ""
+        )}
+        <div
+          class="gauge"
+          style={{
+            "box-shadow":
+              (this.minorSize <= 300 && this.thickness <= 25) ||
+              !this.styleShadow
+                ? "none"
+                : "",
+            height: `${this.minorSize * 0.792 -
+              this.calcThickness() * (this.minorSize / 100)}px`,
+            width: `${this.minorSize * 0.792 -
+              this.calcThickness() * (this.minorSize / 100)}px`
+          }}
+        >
+          {this.showValue ? (
+            <div
+              style={{
+                "font-size": `${(this.minorSize * 0.795 -
+                  this.calcThickness() / 2 * (this.minorSize / 100)) /
+                  8}px`
+              }}
+            >
+              {this.calcPercentage() > 100
+                ? 100
+                : this.calcPercentage() < 0
+                  ? 0
+                  : `${Math.round(this.calcPercentage())}%`}
+            </div>
+          ) : (
+            ""
+          )}
+        </div>
+      </div>
+    );
+  }
+
+  private renderLine(childRanges) {
+    const divRanges = [];
+    const divRangesName = [];
+    let currentMargin = 0;
+    //////////////////////////////////////////
+    function calcPositionRange(preValue) {
```

Comment:
Not sure of having a function for this.

Reply:
It needs that function to going adding values to each range. It is of critical usage within the drawing process!

Reply:
Yes, I understand, I'm just not sure if having a function for such a simple operation was needed.
Also remember that the function is evaluated every time the method is called, and, where it's called, there's a longer operation, that migth make more sense to encapsulate, to simplify reading and understanding.